OSSM-6063 [DOC] OpenShift Service Mesh Console (OSSMC) plugin Integration (#88446)
openshift/openshift-docs: observability/kiali/ossm-console-plugin-assembly.adoc

:_mod-docs-content-type: ASSEMBLY
[id="ossm-console-plugin-assembly"]
= Using OpenShift Service Mesh Console plugin
include::_attributes/common-attributes.adoc[]
:context: ossm-console-plugin-assembly

toc::[]

The {SMPlugin} extends the {ocp-short-name} web console with a {SMProductShortName} menu and enhanced tabs for workloads and services.

[IMPORTANT]
====

> If your browser does not trust the certificate, you can manually accept it in the Kiali standalone UI before accessing the {SMPluginShort}.

Author: @ferhoyos  Can you elaborate on this too? :)

> If your browser does not trust the certificate, you can manually accept it in the Kiali standalone UI before accessing the {SMPluginShort}.

Reviewer: After reviewing with @jmazzitelli, we found that this certificate section is outdated and can be removed from the doc.

====

include::modules/ossm-about-console-plugin.adoc[leveloffset=+1]

:_mod-docs-content-type: PROCEDURE
[id="ossm-install-console-plugin_{context}"]
== Installing {SMPlugin-s}

> You can install the {SMPluginShort} by using the {ocp-short-name} web console or the {ocp-short-name} CLI.
> 
> [NOTE]
> ====
> {SMPluginShort} is only supported on {ocp-short-name} 4.15 and above. For {ocp-short-name} 4.14 users, only the standalone Kiali console is accessible.
> ====

Reviewer: ```suggestion
The OSSMC plugin is installed through the Kiali Operator by creating a `OSSMConsole` resource with the corresponding plugin settings.

### OSSM version compatibility

The OSSMC plugin follows a different versioning scheme than OSSM, so it is necessary to determine the appropriate OSSMC plugin version to install based on the OSSM version:

| OSSM Version    | Kiali Server Version | OSSMC Plugin Version | OCP Version |
| --------------- | -------------------- | -------------------- | ----------- |
| 3.0             | v2.4                 | v2.4                 | 4.15+       |
| 2.6             | v1.73                | v1.73                | 4.14 - 4.18 |
| 2.5             | v1.73                | v1.73                | 4.14 - 4.18 |

[NOTE]
====
{SMPluginShort} is only supported on {ocp-short-name} 4.15 and above. For {ocp-short-name} 4.14 users, only the standalone Kiali console is accessible.
====

[NOTE]
====
In most cases, it is recommended to install the latest version of the Kiali Operator, even when installing a previous OSSMC plugin version, as it includes the latest z-stream releases for all supported OSSMC plugin versions.
====

You can install the {SMPluginShort} by using the {ocp-short-name} web console or the {ocp-short-name} CLI.
```

Author: @ferhoyos I have divided this information between the About OSSMC plugin and the installing section. Do take a look now.

include::modules/ossm-install-console-plugin-ocp-web-console.adoc[leveloffset=+2]

include::modules/ossm-install-console-plugin-ocp-cli.adoc[leveloffset=+2]

> :_mod-docs-content-type: PROCEDURE

Reviewer 2: this doesn't go in assemblies
```suggestion
```

[id="ossm-uninstall-console-plugin_{context}"]
== Uninstalling {SMPlugin-s}

You can uninstall the {SMPluginShort} by using the {ocp-short-name} web console or the {ocp-short-name} CLI.

[NOTE]
====

> You must uninstall the OSSMC plugin before removing the Kiali Operator. Deleting the Operator first may leave OSSMC and Kiali CRs stuck, requiring manual removal of the finalizer. Use the following command with `<CR type>` as `kiali` or `ossmconsole` to remove the finalizer, if needed:
> [source,terminal]
> ----
> $ oc patch <CR type> <CR name> -n <CR namespace> -p '{"metadata":{"finalizers": []}}' --type=merge
> ----

Reviewer 2: user-replaced values are `<lowercase_with_underscores>`. Later, you're using e.g. `custom_resource_name`, should probably stay consistent:
```suggestion
You must uninstall the OSSMC plugin before removing the Kiali Operator. Deleting the Operator first may leave OSSMC and Kiali CRs stuck, requiring manual removal of the finalizer. Use the following command with `<custom_resource_type>` as `kiali` or `ossmconsole` to remove the finalizer, if needed:
[source,terminal]
----
$ oc patch <custom_resource_type> <custom_resource_name> -n <custom_resource_namespace> -p '{"metadata":{"finalizers": []}}' --type=merge
----
```
